lib/std/fmt/parse_float.zig

```diff
@@ -425,4 +248,22 @@ test "fmt.parseFloat" {
             try expect(approxEqAbs(T, try parseFloat(T, "2.71828182845904523536"), @as(T, 2.718281828459045), epsilon));
         }
     }
+
+    // test rounding behavior
+    try expectEqual(@bitCast(u64, try parseFloat(f64, "144115188075855870")), 0x4380000000000000); // exact
+    try expectEqual(@bitCast(u64, try parseFloat(f64, "144115188075855884")), 0x4380000000000000); // round down
+    try expectEqual(@bitCast(u64, try parseFloat(f64, "144115188075855885")), 0x4380000000000000); // round half toward zero
+    try expectEqual(@bitCast(u64, try parseFloat(f64, "144115188075855886")), 0x4380000000000000); // round down??
+    try expectEqual(@bitCast(u64, try parseFloat(f64, "144115188075855889")), 0x4380000000000001); // round up
+    try expectEqual(@bitCast(u64, try parseFloat(f64, "144115188075855900")), 0x4380000000000001); // exact
+
+    try expectEqual(@bitCast(u64, try parseFloat(f64, "9007199254740993")), 0x4340000000000000); // rounded down
+
+    // test precision of f128
+    try expectEqual(@bitCast(u128, try parseFloat(f128, "9007199254740993")), 0x40340000000000000800000000000000); // exact
+
+    // test range of f128
+    // at time of writing (Mar 2021), zig prints f128 values larger than f64 as "inf",
+    // so I'm not 100% sure this hex literal is the corrent parse of 1e4930
+    try expectEqual(@bitCast(u128, try parseFloat(f128, "1e4930")), 0x7ff8136c69ce8adff4397b050cae44c7);
```

Comment:
This looks like it's off by 1 bit to me - should be `0x7ff8136c69ce8adff4397b050cae44c6`

Based on a rudimentary calculation in Python, intended to get the error in ulp:
```python
>>> exp = 0x7ff8 - 0x3fff
>>> (10**4930 - 0x1136c69ce8adff4397b050cae44c7 * 2**(exp-112)) / 2**(exp-112)
-1.2945749668158448
>>> (10**4930 - 0x1136c69ce8adff4397b050cae44c6 * 2**(exp-112)) / 2**(exp-112)
-0.2945749668158449
``` 
I might have made a mistake in the calculation, of course